libnmstate/nm/active_connection.py

```diff
@@ -132,37 +121,12 @@ def _deactivate_connection_callback(self, src_object, result, user_data):
                 % self._nmdev.get_iface()
             )
 
-    def refresh_state(self):
```

Comment:
I do not think that the logic in this method can be dropped. It is mimicking `nmcli` logic.
It states when a connection is 'activating' even though the active-connection states that its state is 'deactivated'.

Note that it also involves the device state as well in the logic.

Perhaps @thom311 can peek in and suggest if this is still needed or not.